### PR TITLE
Fix -L order, LDFLAGS, LD_LIBRARY_PATH issues in GNOME module

### PR DESCRIPTION
**Pull request by lantw44**
These are patches needed to get GNOME in JHBuild works properly on FreeBSD. They are not specific to any operating system, but it is much likely to get build failure caused by these bugs on FreeBSD than on GNU/Linux distributions.

### CODE REVIEW

**Comment by lantw44** [comment]
Ping ... Does someone have time to review my patches? They are required to get meson work with GNOME on FreeBSD.

**Comment by jpakkane** [comment]
This review was stuck in Github UI limbo for a long time. Sorry.

File: mesonbuild/modules/gnome.py [hunk 1]
```diff
@@ -595,6 +596,8 @@ def generate_gir(self, state, args, kwargs):
                 d = os.path.join(state.environment.get_build_dir(), d)
                 scan_command.append('-L' + d)
             scan_command += ['--library', libname]
+        scan_command += filter(lambda s: s.startswith('-L'),
+                               state.environment.coredata.external_link_args[lang])
```
[the remark this answers is not in the document]
Comment (lantw44) [reply mesonbuild/modules/gnome.py]: I am going to replace it with a for loop to make it easier to read.

**Comment by lantw44** [comment]
Ping again ... Can my updated patches get reviewed?

**Pull request merged**